homeassistant/components/starlink/binary_sensor.py

```diff
@@ -0,0 +1,62 @@
+"""Contains sensors exposed by the Starlink integration."""
+
+from __future__ import annotations
+
+from homeassistant.components.binary_sensor import BinarySensorDeviceClass
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import StarlinkUpdateCoordinator
+from .entity import StarlinkBinarySensorEntity, StarlinkBinarySensorEntityDescription
+
+# This stores device_class overrides for alert types that we know aren't "problems".
+DEVICE_CLASS_OVERRIDES = {
+    "alert_is_heating": BinarySensorDeviceClass.HEAT,
+    "alert_install_pending": BinarySensorDeviceClass.UPDATE,
+    "alert_roaming": None,
+    "alert_is_power_save_idle": None,
+}
```

Comment:
Create a tuple of `StarlinkBinarySensorEntity` for each `key` instead

Reply:
Is that preferred, compared to querying Starlink for available alerts? I'm not sure how often alerts change, but when they do we risk having issues that way right?

Reply:
I probably need to give you a bit more color

Here is an example of what I'm thinking:


https://github.com/home-assistant/core/blob/a8cdb86b23dffd0b11bb1bd9f70f28f41a90f7c5/homeassistant/components/august/binary_sensor.py#L124

Reply:
This would essentially be the same as how we set up the sensors platform right?

Reply:
Yes

Reply:
That's what I was referring to earlier - in this case the alerts from Starlink are prone to change. Is it preferred to essentially hardcore them by saying "we will expose a binary sensor for the alert with this tag", versus the current setup that asks Starlink "what is currently available? Let's create binary sensors for them" 

My concern is I'm not sure how often alerts actually change

Reply:
Alters don't sound like they are binary sensors, they should probably be implemented as events

Reply:
I will look at the event docs, but both the app and the API expose these as "problem sensors", though the app also shows banners when there's an alert

Reply:
Just my 2 cents as an observer, but implementing these as events doesn’t seem quite right because they aren’t instantaneous things and instead show the state of the device.  They would be similar to the tesla powerwall integration in how they show if the grid is on, grid services are on, and powerwalls are on.

Reply:
Let's hard code them. If the key is missing you can skip over the sensor